recipes/lora_finetune_single_device.py

```diff
@@ -132,6 +136,10 @@ def __init__(self, cfg: DictConfig) -> None:
         self._resume_from_checkpoint = cfg.resume_from_checkpoint
         self._save_adapter_weights_only = cfg.get("save_adapter_weights_only", False)
         self._gradient_accumulation_steps = cfg.gradient_accumulation_steps
+        self._clip_grad_norm = cfg.get("clip_grad_norm", None)
+        self._log_grad_norm = cfg.get("log_grad_norm", False)
```

Comment:
We don't really need `log_grad_norm`. If `clip_grad_norm` is set, we can log grad norm by default. Computing this each step doesn't increase the time for single device.

Reply:
Yeah based on your results I agree, seems like no negative perf impact to just logging by default. Can you also make the same changes in full_finetune_single_device.py and confirm that no negative perf impact there? (I think we should enable for both recipes in one go.)

Distributed may be a different story if we have to sync, but we can consider that out of scope for this PR.

If the results look similar for full finetune, I agree it makes sense to go ahead and remove the `log_grad_norm` config. I would even bias towards just always logging grad norm (that way you don't have to awkwardly set it to `'inf'` to turn on logging, imo that's a bit unintuitive).